parser/cst.go

```diff
@@ -52,7 +52,7 @@ var (
 			{"HeredocEnd", `\b\1\b`, stateful.Pop()},
 			{"Whitespace", `\s+`, nil},
 			{"Interpolated", `\${`, stateful.Push("Interpolated")},
-			{"Text", `[^\s$]+`, nil},
+			{"Text", `\$|[^\s$]+`, nil},
```

Comment:
Why `\$|[^\s$]` and not `[^\s]`?

Reply:
Because it will then consume the `$` needed for `${`.